Edge cie 4367 monitoring and troubleshooting commands (#1660)
SoftwareAG/c8y-docs: content/edge/operating-edge-bundle/edge-troubleshooting.md

* Check the log files, see [Logging](/edge/operating-edge/#logs-files).

These steps help you analyze the issue and provide a fix. If you need to contact {{< company-sag >}} support, include the output of the diagnostics dump. For more information about using the diagnostics, see [Diagnostic utility](#diagnostics).

### Troubleshooting Microservices {#troubleshooting-microservices}

> In case of any microservices related issues, {{< company-sag >}} recommends you to additionally check the status of the microservices pods using the following command:

Reviewer: @TzviLipschitz Just for my understanding: Does Software AG recommend this or is this a recommendation we (Edge team) make?

Author: This should be Edge Team. I have reworded

	[admin@iot-edge-server ~]$ sudo kubectl get pods -n kube-system

> If any pods show their status as evicted, expand the disk size. See [Expanding the disk size](/edge/operating-edge/#expanding-the-disk-size).

Reviewer: ```suggestion
If any pods show their status as evicted, expand the disk size. For further information see [Expanding the disk size](/edge/operating-edge/#expanding-the-disk-size).
```

Author: This section as been reworked

* Check the log files, see [Logging](/edge/operating-edge/#logs-files).

These steps help you analyze the issue and provide a fix. If you need to contact {{< company-sag >}} support, include the output of the diagnostics dump. For more information about using the diagnostics, see [Diagnostic utility](#diagnostics).

### Troubleshooting Microservices {#troubleshooting-microservices}

> In case of any microservices related issues, it is recommended to:

Reviewer: ```suggestion
In case of any microservices related issues, we recommend you to:
```

Reviewer: @TzviLipschitz I would recommend to remove the bullet list here as you effectively only have 1 bullet point "Check if microservice_hosting...." and the other two bullet points are written as alternatives. So keeping it as a regular paragraph is better.

Otherwise, please re-write point 2 & 3.

> * Check if microservice_hosting is disabled using the command `sudo monit status microservice_hosting`

Reviewer: @TzviLipschitz What does "microservice_hosting" refer to? It might be necessary to format this in some way.

Author: microservice hosting refers to a specific component that can be monitored. monitoring.md has the whole list. I have changed microservice_hosting to be in italics but would "" be more suitable?

![Microservice Hosting Status](/images/edge/monit_status_microservice_hosting.png)

> * If it is showing as disabled, enable the microservice hosting feature through [the GUI](/edge/edge-configuration/#enabling-or-disabling-the-microservice-hosting-feature-using-the-ui) or [the REST API](/api/edge/#tag/Microservice-hosting-feature)

Author: In other places we use the full path for the REST API documentation, i.e. https://cumulocity.com/api/edge/#tag/Microservice-hosting-feature

Is this new way suitable as otherwise older versions will incorrectly link to the latest documentation?

Reviewer: @TzviLipschitz This PR goes into develop, so here it is necessary to keep the new link format (develop always contains the latest status). However, if this is a change that applies to other release versions as well, then the link can be adapted to that release version.

Author: To clarify, by new link format do you mean the full link https://cumulocity.com/api/edge/#tag/Microservice-hosting-feature or the partial link /api/edge/#tag/Microservice-hosting-feature? Would the partial link not work for any version as it would link to the api docs with the same version? I.e if we click the link in the Edge documentation for 10.17 it would link to the api documentation for 10.17

* Check the log files, see [Logging](/edge/operating-edge/#logs-files).

These steps help you analyze the issue and provide a fix. If you need to contact {{< company-sag >}} support, include the output of the diagnostics dump. For more information about using the diagnostics, see [Diagnostic utility](#diagnostics).

### Troubleshooting Microservices {#troubleshooting-microservices}

In case of any microservices related issues, we recommend you to:

Check if _microservice_hosting_ is disabled using the command `sudo monit status microservice_hosting`

![Microservice Hosting Status](/images/edge/monit_status_microservice_hosting.png)
If it is showing as disabled, enable the microservice hosting feature through [the GUI](/edge/edge-configuration/#enabling-or-disabling-the-microservice-hosting-feature-using-the-ui) or [the REST API](/api/edge/#tag/Microservice-hosting-feature).
If enabling the microservice hosting feature fails, check the status of the Kubernetes system pods using the command below:
```shell
sudo kubectl get pods --namespace kube-system
```

> If any pod in the list show its status as evicted, the most probable cause could be that the system ran out of disk space. You should expand the disk size through [the GUI](/edge/operating-edge/#expanding-the-disk-size) or [the REST API](/api/edge/#operation/expandDiskSize). Enable the microservice hosting feature after disk expansion.

Reviewer: ```suggestion
If any pod in the list show its status as evicted, the most probable cause is that the system ran out of disk space. Expand the disk size through [the GUI](/edge/operating-edge/#expanding-the-disk-size) or [the REST API](/api/edge/#operation/expandDiskSize). Enable the microservice hosting feature after disk expansion.
```
